### DISCUSSION TITLE
Is there a template for a .NET 8 Uno Platform Library project?

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [unoplatform/uno#14962] [General]
I'm want to create an Uno Platform Library project that contains multiple user controls.  The current template produces a  .NET 7 library project.  However, I need to use my own .NET 8 libraries, which are not compatible with .NET 7.  

You can't just go into the project file and replace 7's with 8's.  There is the issue of RIDs such as win10-x64 that are no longer compatible with .NET 8 and now need to be win-x64.  

If I were more knowledgeable about template editing, MSBuild, and Nuget packages, I'd offer to create this needed template myself, but I'm not there yet.

Is anyone creating a template for a .NET 8 Uno Platform Library project?

**Comment by limeniye** [comment]
Duplicate of https://github.com/unoplatform/uno/discussions/14927

Hello, @MarkBrainfire ! 
This problem is really important at the moment, but there is no ready-made solution to this problem yet, and I hope that it will appear in the bright future.
This has already been discussed in this thread: https://github.com/unoplatform/uno/discussions/13065#discussioncomment-7614981

Please, subscribe for the following issue: https://github.com/unoplatform/uno.templates/issues/503
(You can also set some emoticon to prioritize this issue)

Now regarding the problem you described.
This problem occurs because UnoLibrary knows nothing about the existence of the Directory.Packages.props file.
Therefore, when creating a project, there is a difference in package versions.
To resolve this issue, follow these steps:
- Place your UnoLibrary project in a directory that has a Directory.Packages.props file.
![image](https://github.com/unoplatform/uno/assets/69314237/def1d9f5-effe-46ca-a58b-208d60cb7a62)

- Delete the package versions in your UnoLibrary.csproj file.
![image](https://github.com/unoplatform/uno/assets/69314237/2b91a471-896f-4b68-8bad-cb4a9e78f500)

**Comment by MarkBrainfire** [comment]
THANK YOU @limeniye I greatly appreciate you taking the time to help me on this!

**Reply by limeniye** [reply]
I'm glad I could help you!

**Comment by MarkBrainfire** [comment]
@limeniye Your solution worked great.  I just have to:

1)  start with a .NET 8 App, but that's no problem, I'll create a .NET 8 solution that demos my UserControls
2) As you suggested, locate my library project in the same directory as the Directory.Packages.props file (the app project directory).
3) As you suggested, in my library project file remove the three package references your red arrows point to above.
4) Replace all the "7"s with "8"s in the library project file.

All a bit goofy ha ha but it will suffice until a more elegant solution is worked out.  Something as obvious as .NET 8 support for Library projects is bound to be a top priority for @jeromelaban and his superb Uno Platform team.  

I like your Wizard suggestion here: https://github.com/unoplatform/uno.templates/issues/503

Uno Platform is to be complimented on the tremendous customizable Wizard for creating apps.  It makes it a breeze to setup projects the way you want.  Now if they can only do the same for Library Projects, it will be awesome!

Mark

**Comment by MarkBrainfire** [comment]
@limeniye In addition to the 4 steps above, I believe it is also necessary to add the following Nuget packages to the .NET 8 dependency group, right? 
![packages](https://github.com/unoplatform/uno/assets/78389339/799decf3-37ae-491e-97cc-d8dc95b12f0d)

And doing so creates the following ItemGroup in library project file:
![image](https://github.com/unoplatform/uno/assets/78389339/6c7f31c2-98bb-48ed-a8ff-bc7edb3afb42)

**Reply by limeniye** [reply]
You can add anything you want. It already depends on your needs.
